lib/include/openturns/OTishigamiusecase.hxx

```diff
@@ -0,0 +1,176 @@
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_OTISHIGAMIUSECASE_HXX
+#define OPENTURNS_OTISHIGAMIUSECASE_HXX
+
+#define _USE_MATH_DEFINES
+#include <cmath>          // For M_PI
+
+#include "openturns/OTconfig.hxx"
+#include "openturns/Distribution.hxx"
+#include "openturns/SymbolicFunction.hxx"
+#include "openturns/Function.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+class IshigamiUseCase
+{
+  public:
+    /* Default constructor */
+    IshigamiUseCase() : dimension_(3), a_(7.0), b_(0.1)
+    {
+      // Create the Ishigami function
+      Description inputVariables(dimension_);
+      inputVariables[0] = "xi1";
+      inputVariables[1] = "xi2";
+      inputVariables[2] = "xi3";
+      Description formula(1);
+      formula[0] = (OSS() << "sin(xi1) + (" << a_ << ") * (sin(xi2)) ^ 2 + (" << b_ << ") * xi3^4 * sin(xi1)");
+      model_ = SymbolicFunction(inputVariables, formula);
+  
+      // Create the input distribution
+      Collection<Distribution> marginals(dimension_);
+      marginals[0] = Uniform(-M_PI, M_PI);
+      marginals[1] = Uniform(-M_PI, M_PI);
+      marginals[2] = Uniform(-M_PI, M_PI);
+      inputDistribution_ = JointDistribution(marginals);
+      
+      // Mean, variance
+      mean_ = a_ / 2;
+      variance_ = pow(b_, 2.0) * pow(M_PI, 8.0) / 18.0 \
+        + b_ * pow(M_PI, 4.0) / 5.0 + pow(a_, 2.0) / 8.0 + 1.0 / 2.0;
+      // Sobol' indices
+      // First order Sobol' index of X1
+      s1_ = (b_ * pow(M_PI, 4.0) / 5.0 + pow(b_, 2.0) * pow(M_PI, 8.0) / 50.0 \
+        + 1.0 / 2.0) /        variance_;
+      s2_ = (pow(a_, 2.0) / 8.0) / variance_;  // First order of X2
+      s3_ = 0.0;  // First order of X3
+      s12_ = 0.0;  // Interaction of (X1, X2)
+      // Interaction of (X1, X3)
+      s13_ = pow(b_, 2.0) * pow(M_PI, 8.0) * (1.0 / 9.0 - 1.0 / 25.0) / 2.0 / variance_;
+      s23_ = 0.0;  // Interaction of (X2, X3)
+      s123_ = 0.0;  // Interaction of (X1, X2, X3)
+      sT1_ = s1_ + s13_;  // Total of X1
+      sT2_ = s2_;  // Total of X2
+      sT3_ = s3_ + s13_;  // Total of X3
+      
+    }
+    Function getModel()
+    {
+      return model_;
+    }
+    UnsignedInteger getDimension()
+    {
+      return dimension_;
+    }
+    Scalar getA()
+    {
+      return a_;
+    }
+    Scalar getB()
+    {
+      return b_;
+    }
+    JointDistribution getInputDistribution()
+    {
+      return inputDistribution_;
+    }
+    Scalar getMean()
+    {
+      return mean_;
+    }
+    Scalar getVariance()
+    {
+      return variance_;
+    }
+    Point getFirstOrderSobolIndices()
+    {
+      const Point firstOrderIndices({s1_, s2_, s3_});
+      return firstOrderIndices;
+    }
+    Point getTotalSobolIndices()
+    {
+      const Point totalIndices({sT1_, sT2_, sT3_});
+      return totalIndices;
+    }
+    Scalar getFirstOrderInteractionSobolIndex(const Indices indices)
+    {
+      Scalar value = -1.0;
```

Comment:
Maybe a test to see if indices is in increasing order? Or you sort indices first, before to go through the tests?

Reply:
Ok. Fixed at 4578d84604266a4e6d4eb217f0aec3dfd83bc6bb